test/Streamly/Test/Internal/Data/Parser.hs

```diff
@@ -0,0 +1,305 @@
+module Main (main) where
+
+import qualified Streamly.Internal.Data.Parser as P
+import qualified Streamly.Internal.Prelude as S
+import qualified Streamly.Internal.Data.Fold as FL
+
+-- import Data.List (partition)
+
+import Test.Hspec(hspec, describe)
+import Test.Hspec.QuickCheck
+import Test.QuickCheck (forAll, chooseInt, Property, property, listOf, vectorOf, (.&&.))
+
+-- Accumulator Tests
```

Comment:
Refactor the number of elements or the list length (10000) that's used in every function to a top level variable like `max_value` or `max_elements` like in other test files.